release-26.2: sql/row: TestJobBackedSeqChunkProvider failed

sql/row.TestJobBackedSeqChunkProvider [failed](https://mesolite.cluster.engflow.com/invocations/default/78cb1b93-a067-4283-8150-e69e44db8ff9?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL3Jvdzpyb3dfdGVzdA==) on release-26.2 @ [1040cbd1b8db50785f373d9d49725eed7d597e1b](https://github.com/cockroachdb/cockroach/commits/1040cbd1b8db50785f373d9d49725eed7d597e1b):

Fatal error:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x81e63e6]

```
Stack: 

```
goroutine 7166 [running]:
github.com/cockroachdb/cockroach/pkg/sql/importer.(*importResumer).prepareTableForIngestion(_, {_, _}, {_, _}, {{0x0, {0x0, 0x0}, 0x0, 0x0, ...}, ...}, ...)
	pkg/sql/importer/import_job.go:591 +0x1a6
github.com/cockroachdb/cockroach/pkg/sql/importer.(*importResumer).Resume.func2({0xac9ac38, 0xc01455d9e0}, {0xad23c98, 0xc0103ab400}, 0xc002acb520)
	pkg/sql/importer/import_job.go:228 +0x19c
github.com/cockroachdb/cockroach/pkg/sql.DescsTxn.func1({0xac9ac38, 0xc01455d9e0}, {0xad288a8, 0xc0103ab400})
	pkg/sql/exec_util.go:3636 +0x84
github.com/cockroachdb/cockroach/pkg/sql.DescsTxn.(*InternalDB).DescsTxn.func2({0xac9ac38, 0xc01455d9e0}, 0xc0103ab400)
	pkg/sql/internal.go:1960 +0x53
github.com/cockroachdb/cockroach/pkg/sql.(*InternalDB).txn.func4({0xac9ac38, 0xc01455d9e0}, 0xc0114a16c0)
	pkg/sql/internal.go:2063 +0x4d4
github.com/cockroachdb/cockroach/pkg/kv.(*Txn).exec(0xc0114a16c0, {0xac9ac38, 0xc01455d9e0}, 0xc001b94940)
	pkg/kv/txn.go:1172 +0xae
github.com/cockroachdb/cockroach/pkg/kv.runTxn({0xac9ac38, 0xc01455d9e0}, 0xc0114a16c0, 0xc001b94940)
	pkg/kv/db.go:1165 +0x6c
github.com/cockroachdb/cockroach/pkg/kv.(*DB).TxnWithAdmissionControl(0xc0031e0c60, {0xac9ac38, 0xc01455d9e0}, 0x0, 0x0, 0x0, 0xc001b94940)
	pkg/kv/db.go:1125 +0x1e5
github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn(...)
	pkg/kv/db.go:1097
github.com/cockroachdb/cockroach/pkg/sql.(*InternalDB).txn(0xc00cb78990, {0xac9ac38, 0xc01455d9e0}, 0xc0107e9da0, {0x0, 0x0, 0xc010d024b0?})
	pkg/sql/internal.go:2050 +0x6d2
github.com/cockroachdb/cockroach/pkg/sql.(*InternalDB).DescsTxn(...)
	pkg/sql/internal.go:1958
github.com/cockroachdb/cockroach/pkg/sql.DescsTxn({0xac9ac38, 0xc01455d9e0}, 0xc0027f8308, 0xc010d024b0, {0x0, 0x0, 0x0})
	pkg/sql/exec_util.go:3635 +0x16d
github.com/cockroachdb/cockroach/pkg/sql/importer.(*importResumer).Resume(0xc013c82380, {0xac9ac38, 0xc01455d9e0}, {0x94a9ca0, 0xc00ed60628})
	pkg/sql/importer/import_job.go:225 +0x513
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine.func2(0xc001bcfcc0, 0xc0050f0008, 0xc013b99720, {0xac94b20, 0xc013c82380}, {0xac9ac38, 0xc01455d9e0}, {0x94a9ca0, 0xc00ed60628})
	pkg/jobs/registry.go:1661 +0x16b
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine(0xc0050f0008, {0xac9c4f0, 0xc0142ce460}, {0x94a9ca0, 0xc00ed60628}, {0xac94b20, 0xc013c82380}, 0xc002bf9500, {0xc00e89f299, 0x7}, ...)
	pkg/jobs/registry.go:1662 +0x10ff
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob(0xc0050f0008, {0xac9c4f0, 0xc0142ce460}, {0xac94b20, 0xc013c82380}, 0xc002bf9500, {0xc00e89f299, 0x7}, {0xc0119ee4e0, 0x17})
	pkg/jobs/adopt.go:503 +0x830
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resumeJob.func1({0xac9c4f0, 0xc011c6ebe0})
	pkg/jobs/adopt.go:350 +0x199
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func1({0xac9c4f0, 0xc011c6ebe0})
	pkg/util/stop/stopper.go:447 +0x112
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 7107
	pkg/util/stop/stopper.go:445 +0x247
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestJobBackedSeqChunkProvider
I260420 09:45:33.125859 1 (gostd) rand.go:288  [-] 1  random seed: -4698750432063917217
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestJobBackedSeqChunkProvider2732899281
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    test_server_shim.go:638: DRPC is enabled (via testserver factory defaults)
=== RUN   TestJobBackedSeqChunkProvider/row-in-first-chunk
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestJobBackedSeqChunkProvider.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-63091

--- a/pkg/sql/row/BUILD.bazel
+++ b/pkg/sql/row/BUILD.bazel
@@ -146,6 +146,7 @@ go_test(
         "//pkg/testutils/skip",
         "//pkg/testutils/sqlutils",
         "//pkg/testutils/testcluster",
+        "//pkg/upgrade/upgradebase",
         "//pkg/util/encoding",
         "//pkg/util/hlc",
         "//pkg/util/leaktest",

--- a/pkg/sql/row/expr_walker_test.go
+++ b/pkg/sql/row/expr_walker_test.go
@@ -22,6 +22,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/row"
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/eval"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
+	"github.com/cockroachdb/cockroach/pkg/upgrade/upgradebase"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/protoutil"
@@ -76,7 +77,20 @@ func TestJobBackedSeqChunkProvider(t *testing.T) {
 	defer log.Scope(t).Close(t)
 
 	ctx := context.Background()
-	srv, sqlDB, kvDB := serverutils.StartServer(t, base.TestServerArgs{})
+	srv, sqlDB, kvDB := serverutils.StartServer(t, base.TestServerArgs{
+		Knobs: base.TestingKnobs{
+			// Prevent the registry from adopting the mock import job, which has
+			// empty ImportDetails and would crash in prepareTableForIngestion.
+			JobsTestingKnobs: &jobs.TestingKnobs{
+				DisableAdoptions: true,
+			},
+			// DisableAdoptions needs this.
+			UpgradeManager: &upgradebase.TestingKnobs{
+				DontUseJobs:                       true,
+				SkipJobMetricsPollingJobBootstrap: true,
+			},
+		},
+	})
 	defer srv.Stopper().Stop(ctx)
 	s := srv.ApplicationLayer()
 

